package/MDAnalysis/analysis/dielectric.py

```diff
@@ -0,0 +1,148 @@
+# MDAnalysis: A Toolkit for the Analysis of Molecular Dynamics Simulations.
+# J. Comput. Chem. 32 (2011), 2319--2327, doi:10.1002/jcc.21787
+#
+
+
+r"""
+Dielectric --- :mod:`MDAnalysis.analysis.dielectric`
+===========================================================
+
+:Authors: Mattia Felice Palermo, Philip Loche
+:Year: 2019
+:Copyright: GNU Public License v2
+
+A tool to compute the average dipole moment :math:`M` and
+the static dielectric constant
+
+.. math::
+
+   \varepsilon = 1 + \frac{\langle M^2 \rangle - \langle M \rangle^2}
+                            {3 \varepsilon_ 0 V k_B T}
+
+for a system simulated in tin foil boundary conditions.
+"""
+
+from __future__ import division
+
+import numpy as np
+from numpy.testing import assert_almost_equal
+
+from MDAnalysis.units import constants, convert
+from MDAnalysis.analysis.base import AnalysisBase
+from MDAnalysis.lib.mdamath import make_whole
+from MDAnalysis.due import due, Doi
+
+due.cite(Doi("10.1080/00268978300102721"),
+description="Dielectric analysis",
+path="MDAnalysis.analysis.dielectric",
+cite_module=True)
+del Doi
+
+class DielectricConstant(AnalysisBase):
+    r"""Dielectric constant
+
+    Parameters
+    ----------
+    selection : AtomGroup
+          any atomgroup
+    temperature : float
+        Temperature (Kelvin) at which the system has been simulated [298.15]
+    make_whole : bool
+        Make molecules whole; If the input already contains whole molecules
+        this can be disabled to gain speedup; the default is ``True``
+    verbose : bool
+          Show detailed progress of the calculation if set to ``True``; the
+          default is ``False``
+
+    Attributes
+    ----------
+    results : dict
+          * M: Directional dependant dipole moment
+            :math:`\langle \boldsymbol M \rangle` in :math:`eÅ`.
+          * M2: Directional dependant squared dipole moment
+            :math:`\langle \boldsymbol M^2 \rangle` in :math:`(eÅ)^2`
+          * fluct: Directional dependant dipole moment fluctuation
+            :math:`\langle \boldsymbol M^2 \rangle - \langle \boldsymbol M \rangle^2`
+            in :math:`(eÅ)^2`
+          * eps: Directional dependant static dielectric constant
+          * eps_mean: Static dielectric constant
+
+    Example
+    -------
+    Create a DielectricConstant object by supplying a selection,
+    then use the :meth:`run` method::
+
+      diel = DielectricConstant(selection)
+      diel.run()
+
+    The static dielectric constant of the system will be returned
+    within the `results` variable of the object.
+
+
+    .. versionadded:: 0.20.0
+
+    """
+    def __init__(self, selection, temperature=300, make_whole=True, **kwargs):
+        super(DielectricConstant, self).__init__(selection.universe.trajectory,
+                                                 **kwargs)
+        self.selection = selection
+        self.temperature = temperature
+        self.charges = selection.charges
```

Comment:
If there is no charge defined in the topology, this will likely raise a not so useful `AttributeError`. It would be nice to catch it, and raise a `NoDataError` with a friendly message, instead.

Reply:
Good idea. I added this and also a check if the total charge is non zero